src/runtime/thread_pool.cc

```diff
@@ -285,19 +315,37 @@ class ThreadPool {
   // Initialize the pool.
   void Init() {
     for (int i = 0; i < num_workers_; ++i) {
+      // The SpscTaskQueue only host ONE item at a time
       queues_.emplace_back(
-          std::unique_ptr<ParallelTaskQueue>(new ParallelTaskQueue()));
+          std::unique_ptr<SpscTaskQueue>(new SpscTaskQueue()));
     }
     threads_.resize(num_workers_);
     for (int i = 0; i < num_workers_; ++i) {
       threads_[i] = std::thread([this, i] {
           this->RunWorker(queues_[i].get());
         });
     }
+    const char *val = getenv("TVM_BIND_THREADS");
+    if (val == nullptr || atoi(val) == 1) {
+      uint32_t num_cores;
+#if defined(_M_X64) || defined(__x86_64__)
+      // Half to not count hyper threading.
+      num_cores = std::thread::hardware_concurrency() / 2;
+#else
+      num_cores = std::thread::hardware_concurrency();
+#endif
+      if (num_workers_ <= num_cores) {
```

Comment:
Add an env variable TVM_BIND_THREADS, which default to 1, and set affinity when it is set to 1

Reply:
OK this is already here